### ISSUE TITLE
Every shipped detector metric is a single seed — obituary recall at 0.85 spans 0.6599-0.8081 with everything else fixed

### ISSUE THREAD

**Issue by ducroq** [veen-systems/llm-distillery#158]
Measured 2026-09-10 (`EXP-033`, `EXP-034`). Evidence: `docs/evidence/2026-09-10-detector-window-128-vs-512/`, `docs/evidence/2026-09-10-obituary-version-ordering-multiseed/`.

## The defect

The detector trainers (`filters/common/obituary_detector/training/train_v1.py`, `filters/common/violence_promotion/training/train_v1.py`) build the head as:

```python
MLPClassifier(hidden_layer_sizes=(256,128), max_iter=400,
              early_stopping=True, n_iter_no_change=15, random_state=SEED)
```

`early_stopping=True` makes `random_state` choose the internal validation split. With corpus, embedder, window and hyperparameters all held fixed, **obituary heldout recall at the live 0.85 op-point ranges 0.6599 to 0.8081 across five seeds** — a 0.148 spread, and 0.285 at threshold 0.95.

**`SEED = 42` is hardcoded, so every published detector number is one draw from that distribution, not a property of the model.**

## Why it matters beyond bookkeeping

`EXP-034` retrained v3/v4/v5 from their own corpora across five seeds. All three pairwise recall bands overlap at 0.85/0.90/0.95, and the versions finish in **three different orders** across the five seeds — every version wins under some seed. A version comparison built on single-seed numbers had propagated into a downstream repo's dependency doc as settled, and reversed direction once seeds were varied.

## What would close this

Not a code change on its own — a reporting rule:

1. Any detector comparison averages over a seed set and reports the **band**, not a point estimate. `EXP-033`'s set (42, 7, 13, 101, 2026) is a reasonable default.
2. `training_config.json` records the seed set and the band, not one number.
3. Optional and cheaper than it sounds: fit N heads and keep the median-recall one, or ensemble them. Untested here.

## Not claimed

- This does **not** say the shipped models are bad, or that v5 should be replaced. v5 stays live at 0.85; `EXP-034` gives no reason to change it.
- Seed and training data are **not** separated in `EXP-034` — each version kept its own corpus, so v4's 12 hard negatives and v5's 21 hard positives stay confounded with the seed. Not attempted.
- A refit at seed 42 correlates 0.9771/0.9713/0.9952 with the shipped models but is **not** them (sklearn `MLPClassifier` is not bit-reproducible across environments), so this measures the distribution a corpus+protocol produces, not a reproduction.

**Comment by ducroq** [comment]
## The cost argument for multi-seed just moved — b650 now has an RTX 5090

b650's RTX 3090 Ti was replaced by an **RTX 5090 32 GB** on 2026-09-17. Measured on the new card: the `uplifting v7` student scores **660 rows in 12 s** on CUDA, against ~2 min recorded for the 3090 Ti and **9 min 13 s** on the same box's CPU. VRAM went 24 → 32 GB.

That is an **inference** figure and I am not going to dress it up as a training one — training throughput on the new card has not been timed. But it removes the cheap half of the objection to running detector metrics across seeds: the *evaluation* passes that a seed sweep multiplies are now close to free.

⚠️ **One caveat that cuts the other way, and it is specific to this issue.** Whether a model *trained* on the 5090 is comparable to one trained on the 3090 Ti is **unmeasured** — `H-DEV-1` in `memory/hypothesis-ledger.md`. A 2026-07-09 gotcha already records that the same seed and a fresh re-train give different weights on one box, so an architecture change has no reason to be gentler. For this issue that matters twice over:

1. **Every shipped detector checkpoint was trained on the old card.** A seed sweep run now on the 5090 produces a band that may not contain the shipped model's own behaviour, and the gap would be attributed to seed when part of it is silicon.
2. The honest ordering is the cheap same-box step first — **train twice at one fixed seed on the 5090 and diff the weights.** If same-box training is not reproducible, the seed band swallows the architecture question and there is nothing to separate. Only if it *is* reproducible does comparing across cards mean anything.

So: the box is available and much faster, and the sweep is worth doing — but design it so seed and hardware are not varied at once, which is the confound the 2026-08-10 parity decomposition had to undo once already.

Related: #85 (obituary v6 retrain on b650, same caveat), #104, `EXP-038`.

**Comment by ducroq** [comment]
## Mechanized, and the mechanism has now produced artifacts (`EXP-040`, 2026-09-17, $0)

**The rule is in code, not prose.** `filters/common/detector_seeds.py` owns the seed set; both
detector trainers lost `SEED = 42`, run out-of-fold **once per seed** over `EXP-033`'s set, and
write `seed_set` + a `<metric>_band` beside every point metric. ⛔ **The ARTIFACT seed stays 42 on
purpose** — this is a REPORTING defect, and changing which head ships would quietly make it a model
change. Commits `842e39e`, `45a17c7`, `bb06760`.

### The three parts of "what would close this", against what shipped

1. **Comparisons average over a seed set and report the band** — the trainers now do; enforced by
   `scripts/verification/check_detector_metric_bands.py` + `tests/unit/test_detector_metric_bands.py`
   (18 tests), seeded red on the real tree.
2. **`training_config.json` records the seed set and the band** — done, and at **two sites**:
   obituary's `models/` is gitignored, so `v{3,4,5}/models/training_config.json` exists only on the
   boxes that built it while `v{N}/calibration_report.json` is TRACKED and embeds the same numbers.
   Checking only `models/` would have left **4 of the 9 sites** unguarded, i.e. every copy a reader
   can actually open.
3. **Fit N heads and keep the median / ensemble** — ⛔ **not done, deliberately.** That changes the
   shipped head; this issue says it does not claim the shipped models are bad.

### The nine pre-#158 sites are declared, not retrained

Each carries `single_seed` naming the seed, this issue, and why it was left alone. ⚠️ **Commerce v2's
says `UNRECORDED`, not 42**: its config declares an `MLPClassifier` while the trainer in the tree
builds a transformers `Trainer`, so the shipped head did not come from that path and a 42 there
would be a hand-built fact.

### ⭐ And the bands are now measured, not asserted (`EXP-040`)

At commit `842e39e` the checker read **`0 banded`** — in place, never fired. Both trainers were then
run on b650's RTX 5090 (scratch dirs; **nothing shipped was touched**) and it reads **2 banded,
0 declared** per detector. Both trainers, because only the obituary edit had ever executed and two
files changed in one commit are not one change.

| obituary, `v5_train_seed.jsonl` (11,329 rows), 5-fold OOF | min | median | max | spread | seed 42 |
|---|---|---|---|---|---|
| `oof_recall_at_0.85` | 0.7275 | 0.7565 | 0.7873 | 0.0598 | **0.7275 — the minimum** |
| `oof_recall_at_0.95` | 0.6184 | 0.6778 | 0.7361 | 0.1177 | **0.6184 — the minimum** |

**Seed 42 — the seed every shipped number was published at — is the worst of the five at both
thresholds measured.** ⚠️ Not a general law: on the violence corpus seed 42 sits at the **median**
(0.5498 of 0.5403–0.5829) and **reproduces the shipped figure exactly**, same corpus and protocol,
different box, seven weeks later. The obituary arm does **not** reproduce its shipped 0.6002 (got
0.6184), which is what stops the exact match being tautological — `EXP-034` already measured that a
refit correlates with but is not the shipped head.

⛔ **These are OUT-OF-FOLD numbers on the training corpus. This issue quotes HELDOUT recall
(0.6599–0.8081).** Different population, different instrument — neither reproduces the other, and
the evidence README says so above its first table.

### Left open

**The heldout band.** It is the number this issue actually quotes and the one that would give the
live 0.85 op-point a defensible range. Cheap on the 5090 (embed + 5 fits per detector) and not run.

Evidence: `docs/evidence/2026-09-17-detector-seed-bands/`. Registry `EXP-040`.
Mechanized row in `memory/gotcha-log.md`.
